test/box-luatest/gh_7897_sending_arbitrary_iproto_packets_test.lua

```diff
@@ -0,0 +1,448 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new{
+        alias   = 'dflt',
+    }
+    cg.server:start()
+    cg.server:exec(function()
+        box.session.on_connect(function()
+            rawset(_G, 'sid', box.session.id())
+        end)
+    end)
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.before_each(function(cg)
+    cg.server:exec(function(net_box_uri)
+        local uri = require('uri')
+        local socket = require('socket')
+        local t = require('luatest')
+
+        -- Connect to the server.
+        local u = uri.parse(net_box_uri)
+        local s = socket.tcp_connect(u.host, u.service)
+        t.assert_is_not(s, nil)
+        -- Skip the greeting.
+        t.assert_equals(#s:read(128), 128)
+        rawset(_G, 's', s)
+    end, {cg.server.net_box_uri})
+end)
+
+-- Checks that `box.iproto.send` works correctly.
+g.test_box_iproto_send = function(cg)
+    cg.server:exec(function()
+        local msgpack = require('msgpack')
+        local t = require('luatest')
+
+        local sid = rawget(_G, 'sid')
+        local s = rawget(_G, 's')
+
+        local lua_header = {a = 1}
+        local lua_body = {b = 1}
+        local mp_header = msgpack.encode(lua_header)
+        local mp_body = msgpack.encode(lua_body)
+
+        local mp
+        local mp_len
+        local packet_size
+        local packet_len
+        local packet_header
+        local packet_body
+        local next
+
+        -- Checks that packet consisting only of MsgPack header is sent
+        -- correctly.
+        box.iproto.send(sid, mp_header)
+        packet_size = 5 + #mp_header
+        mp = s:read(packet_size)
+        mp_len = #mp
+        t.assert_equals(mp_len, packet_size)
+        packet_len, next = msgpack.decode(mp)
+        t.assert_equals(mp_len - next + 1, packet_len)
+        packet_header, next = msgpack.decode(mp, next)
+        t.assert_equals(next, mp_len + 1)
+        t.assert_equals(packet_header, lua_header)
+
+        -- Checks that packet consisting only of Lua header is sent correctly.
+        box.iproto.send(sid, lua_header)
+        packet_size = 5 + #mp_header
+        mp = s:read(packet_size)
+        mp_len = #mp
+        t.assert_equals(mp_len, packet_size)
+        packet_len, next = msgpack.decode(mp)
+        t.assert_equals(mp_len - next + 1, packet_len)
+        packet_header, next = msgpack.decode(mp, next)
+        t.assert_equals(next, mp_len + 1)
+        t.assert_equals(packet_header, lua_header)
```

Comment:
Note (can resolve yourself): I see this part repeated in the test quite many times, could be extracted into a function. But I trust Vladimir already checked it all and it was kept for simplicity or something.

Reply:
> But I trust Vladimir already checked it all and it was kept for simplicity or something.

I did not. I usually only skim through tests. If there's a lot of code duplication that could be eliminated by adding a helper function, I'm all for it.

Reply:
You are surely right, TBH, I was simply lazy to refactor the common testing part. Thanks for insisting on this! Added a `test_iproto_send` helper.